### PR TITLE
Fix  : No snackbar when sync button is clicked for the first time

### PR DESCRIPTION
**Pull request by sidsharma2002**
#888 Initially the flow was wrong because the starting value of _isSyncing is null. When we open the next time,   unexpectedly the worker emits a work with succeeded state , the null check in the when block prevents the snackbar to show but sets its value as false. Then, when we click the sync button everything happens as intended (due to the false value set).

### CODE REVIEW

File: app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardViewModel.kt [hunk 1]
```diff
@@ -70,60 +66,44 @@ class DashboardViewModel(
...
-                }
-
+                showSnackBar?.invoke(R.string.sync_preparing)
                 _isSyncing = true
             }
         }
     }
 
     init {
-        updateData()
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardViewModel.kt:119]: This method was here to refresh the screen when going back to it after capturing trees. We still need it.

**Review by Elforama — CHANGES_REQUESTED** [review]
Great work, it looks like the issue is resolved.

There is just one issue. When capturing trees before the user syncs, the dashboard doesn't show the tree count. Adding back the updateData method seems to fix it.

**Review by Elforama — APPROVED** [review]
Awesome job, it works well!

Please fix any conflicts and I'll get this merged.

**Review by ZavenArra — APPROVED** [review]
Tested and works well.

**Pull request merged**